src/test/java/com/iexec/sms/secret/SecretControllerTests.java

```diff
@@ -0,0 +1,318 @@
+
+    @Mock
+    private Web3SecretService web3SecretService;
+
+    @InjectMocks
+    private SecretController secretController;
+
+    private static final SecureRandom seed = new SecureRandom();
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    //region isWeb3SecretSet
+    @Test
+    void shouldReturnNoContentWhenWeb3SecretExists() {
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.of(new Web3Secret()));
+        assertThat(secretController.isWeb3SecretSet(WEB3_SECRET_ADDRESS))
+                .isEqualTo(ResponseEntity.noContent().build());
+        verifyNoInteractions(authorizationService, web2SecretsService);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenWeb3SecretDoesNotExist() {
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.empty());
+        assertThat(secretController.isWeb3SecretSet(WEB3_SECRET_ADDRESS))
+                .isEqualTo(ResponseEntity.notFound().build());
+        verifyNoInteractions(authorizationService, web2SecretsService);
+    }
+    //endregion
+
+    //region getWeb3Secret
+    @Test
+    void failToGetWeb3SecretWhenBadAuthorization() {
+        when(authorizationService.getChallengeForGetWeb3Secret(WEB3_SECRET_ADDRESS))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(false);
+        assertThat(secretController.getWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, false))
+                .isEqualTo(ResponseEntity.status(HttpStatus.UNAUTHORIZED).build());
+        verifyNoInteractions(web2SecretsService, web3SecretService);
+    }
+
+    @Test
+    void failToGetWeb3SecretWhenSecretDoesNotExist() {
+        when(authorizationService.getChallengeForGetWeb3Secret(WEB3_SECRET_ADDRESS))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(true);
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS)).thenReturn(Optional.empty());
+        assertThat(secretController.getWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, false))
+                .isEqualTo(ResponseEntity.notFound().build());
+        verifyNoInteractions(web2SecretsService);
+    }
+
+    @Test
+    void getWeb3Secret() {
+        when(authorizationService.getChallengeForGetWeb3Secret(WEB3_SECRET_ADDRESS))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(true);
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS)).thenReturn(Optional.of(new Web3Secret()));
+        assertThat(secretController.getWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, false))
+                .isEqualTo(ResponseEntity.notFound().build());
```

Comment:
```suggestion
        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS, false)).thenReturn(Optional.of(new Web3Secret()));
        assertThat(secretController.getWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, false))
                .isEqualTo(ResponseEntity.noContent().build());
```

Same issue as before, but there the default return value doesn't help us.